axelor-account/src/main/java/com/axelor/apps/account/service/reconcilegroup/ReconcileGroupLetterServiceImpl.java

```diff
@@ -40,13 +47,27 @@ public ReconcileGroupLetterServiceImpl(
 
   @Override
   public void letter(ReconcileGroup reconcileGroup) throws AxelorException {
-
+    Company company = reconcileGroup.getCompany();
     List<MoveLine> moveLines =
         moveLineRepository
             .all()
             .filter("self.reconcileGroup = :reconcileGroup")
             .bind("reconcileGroup", reconcileGroup)
             .fetch();
+    if (CollectionUtils.isEmpty(moveLines)) {
```

Comment:
I think this method will take a long time when we try to reconcile a lot of MoveLines (example : > 100 lines) right ?

Reply:
Yes, it can be better to add a check at the start of the letter method that check moveLines on this reconcileGroup and not the right company.
"self.reconcileGroup = :reconcileGroup AND self.move.company != self.reconcileGroup.company"

Then in the letter method, filter on the reconcileGroup's company only.

For the check, reuse your error message if only one moveLine is wrong, if more than one, create another message with all moveline's name.
Also add the reconcileGroup as model in the exception.